src/ghostwriter/factory.py

```diff
@@ -0,0 +1,124 @@
+"""Component factory and process-wide status for ghostwriter."""
+
+from __future__ import annotations
+
+import structlog
+import yaml
+from safir.slack.webhook import SlackWebhookClient
+from structlog.stdlib import BoundLogger
+
+from .dependencies.config import config_dependency
+from .models.v1.mapping import RouteMap
+from .services.rsp_client_manager import RSPClientManager
+from .storage.gafaelfawr import GafaelfawrManager
+
+__all__ = ["Factory", "ProcessContext"]
+
+
+class ProcessContext:
+    """Per-process application context.
+
+    This object caches all of the per-process singletons that can be reused
+    for every request.
+
+    Attributes
+    ----------
+    base_url
+        Base URL for the application; read from config.
+
+    config
+        Application config.
+
+    mapping
+        Rewrite mapping; read from file specified in config.
+
+    gafaelfawr_manager
+        Cache for token-to-user-and-capability mappings.
+
+    rsp_client_manager
+        Cache for token-to-rsp-client mapping.
+    """
+
+    def __init__(self) -> None:
+        self.logger = structlog.get_logger("ghostwriter")
+        self.config = config_dependency.config
+        self.base_url = self.config.environment_url
+        if self.base_url is None:
+            raise RuntimeError("config.environent_url must be set")
+        self.gafaelfawr_manager = GafaelfawrManager(base_url=self.base_url)
+        self.rsp_client_manager = RSPClientManager(
+            base_url=self.base_url,
+            logger=self.logger,
+            gafaelfawr_manager=self.gafaelfawr_manager,
+        )
+        self.reload_map()
```

Comment:
```suggestion
        self.mapping = None
        self.reload_map()
```
Might make it a bit easier for IDE's & linters if mapping appears in `__init__` even if it's initialized in `reload_map`

Reply:
I have no external usage of reload_map() so if I call it load_map() and have it return the value, then I can initialize and set the object in one step.